pkg/policy/distillery.go

```diff
@@ -49,6 +49,17 @@ func (cache *policyCache) lookupOrCreate(identity *identityPkg.Identity) *cached
 	return cip
 }
 
+// GetPolicySnapshot returns a snapshot of the current policy cache.
+func (cache *policyCache) GetPolicySnapshot() map[identityPkg.NumericIdentity]SelectorPolicy {
+	cache.Lock()
+	defer cache.Unlock()
```

Comment:
I was looking at the mutex usage here to understand if there are potential deadlocks. I didn't find any, i.e. the existing code indeed seems to assume that the lock order is `Repository.mutex` before `policyCache.Mutex`.

However, I did notice that the `cachedPolicy` pointer can be updated even after `cache.Lock()` is held:
https://github.com/cilium/cilium/blob/12565044f0b79d0a11a1fed80440682d6bed50c9/pkg/policy/distillery.go#L80-L90

I assume this is not a problem, i.e. we just assume that that policy update happened before the snapshot (even though technically it could happen during the snapshot, but just for one single identity).

No action required in terms of code review. Documenting the lock order might be nice to have though.